extension/content_script.js

```diff
@@ -584,13 +585,20 @@ function removeUI() {
 
 function toggleExtensionVisibility() {
 	if (timeTrackerView === null) {
+		settings.setAppStatus(true); 
 		setupUI();
 	} else {
 		removeUI();
+		settings.setAppStatus(false); 
 	}
 }
 
 function setupKeyListenerForOnOff() {
+	let appStatus = settings.getAppStatus(); 
```

Comment:
What do you think about:
1. Move this appstatus snippet to oneTimeSetup.
 -- it's not really about setting up key listener, and this logic of "immediately show ui?" would be something i expected to see in oneTimeSetup.

2. im not sure why onetimesetup needs to accept a callback.
Why not just have setupKeyListenerForOnOff be called in line 552?
Then it's clear what methods are exactly being called once per webpage -- just read that one function. No need to figure out what cb() is.

Reply:
Agree, agree